src/plugins/formulas/hyperformulaSetup.js

```diff
@@ -0,0 +1,74 @@
+import staticRegister from '../../utils/staticRegister';
+import hyperformulaDefaultSettings from './hfDefaultSettings';
+
+/**
+ * Registers HyperFormula as a global entity and applies hooks allowing a multi-sheet setup.
+ *
+ * @param {object} pluginSettings Plugin settings.
+ * @param {string} hotId Handsontable guid.
+ * @returns {HyperFormula} The HyperFormula instance.
+ */
+export function registerHF(pluginSettings, hotId) {
+  const HFClass = pluginSettings.hyperformula;
+
+  if (!staticRegister('formulas').hasItem('hyperformulaInstances')) {
+    staticRegister('formulas').register('hyperformulaInstances', new Map());
+  }
+
+  const hfInstances = staticRegister('formulas').getItem('hyperformulaInstances');
+
+  // Register custom functions
+  if (pluginSettings.functions) {
+    pluginSettings.functions.forEach((func) => {
+      const {
+        name,
+        plugin,
+        translations
+      } = func;
+
+      HFClass.registerFunction(name, plugin, translations);
+    });
+  }
+
+  // Register languages
+  if (pluginSettings.language) {
+    const {
+      code,
+      language
+    } = pluginSettings.language;
+
+    HFClass.registerLanguage(code, language);
+  }
+
+  // Create instance
+  const hfInstance = HFClass.buildEmpty(hyperformulaDefaultSettings);
+
+  // Add it to global registry
+  hfInstances.set(hfInstance, [hotId]);
+
+  // Register named expressions
+  if (pluginSettings.namedExpressions) {
+    hfInstance.suspendEvaluation();
```

Comment:
In the selected fragment above I had trouble with `hfInstances` vs `hfInstance`. Maybe we can use `hfRegistry` to make the difference obvious?

Reply:
You're right, that'd be much clearer.